shared/util/saga.js

```diff
@@ -11,6 +12,27 @@ import {isArray} from 'lodash-es'
 
 export type SagaGenerator<Yield, Actions> = Generator<Yield, void, Actions>
 
+class SagaLogger {
+  error: LogFn
+  warn: LogFn
+  info: LogFn
+  debug: LogFn
+  isTagged = false
+  constructor(actionType, fcnTag) {
+    const prefix = `${fcnTag} [${actionType}]:`
+    this.debug = (...args) => logger.debug(prefix, ...args)
+    this.error = (...args) => logger.error(prefix, ...args)
+    this.info = (...args) => logger.info(prefix, ...args)
+    this.warn = (...args) => logger.warn(prefix, ...args)
+  }
+  // call this first in your saga if you want chainAction / chainGenerator to log
+  // before and after you run
+  tag = () => {
```

Comment:
Is this used at all right now?

Reply:
No - it was inspired by the service's log helper which makes lines like these
```
▶ [DEBU keybase conflicts.go:48] 8fc7 + GetConflictInfo(...) 
▶ [DEBU keybase teams.go:395] 8f6f - GetConflictInfo(...) -> ok
```
If we want we can add more to this in the future, like timing and generated tags.